metricflow/model/parsing/dir_to_model.py

```diff
@@ -206,41 +209,28 @@ def parse_config_yaml(
                 materialization_validator.validate(config_document[document_type])
                 results.append(materialization_class.parse_obj(object_cfg))
             else:
-                errors.append(
-                    str(
-                        ParsingException(
-                            message=f"Invalid document type: {document_type}. Expected {DOCUMENT_TYPES}.",
-                            ctx=ctx,
-                            config_filepath=config_yaml.filepath,
-                        )
+                issues.append(
+                    ValidationError(
+                        context=FileContext(file_name=ctx.filename, line_number=ctx.start_line),
+                        message=f"Invalid document type: {document_type}. Expected {DOCUMENT_TYPES}.",
                     )
                 )
-
-        if len(errors) > 0:
-            errors_str = "\n".join([str(x) for x in errors])
-            raise ParsingException(
-                message=f"Found {len(errors)} error(s) parsing configs files:\n"
-                f"{textwrap.indent(errors_str, prefix='    ')}"
-            )
     except exceptions.ValidationError as e:
-        raise ParsingException(
-            message=f"YAML file did not conform to metric spec.\nError: {e}",
-            ctx=ctx,
-            config_filepath=config_yaml.filepath,
-        ) from e
+        context = FileContext(file_name=ctx.filename, line_number=ctx.start_line) if ctx is not None else None
+        issues.append(
+            ValidationError(
+                context=context,
+                message=f"YAML document did not conform to metric spec.\nError: {e}",
+            )
+        )
     except ScannerError as e:
-        raise ParsingException(
-            message=str(e),
-            ctx=ctx,
-            config_filepath=config_yaml.filepath,
-        ) from e
-    except ParsingException:
-        raise
+        context = FileContext(file_name=ctx.filename, line_number=ctx.start_line) if ctx is not None else None
+        issues.append(ValidationError(context=context, message=str(e)))
+    except ParsingException as e:
+        context = FileContext(file_name=ctx.filename, line_number=ctx.start_line) if ctx is not None else None
+        issues.append(ValidationError(context=context, message=str(e)))
     except Exception as e:
-        raise ParsingException(
-            message=str(e),
-            ctx=ctx,
-            config_filepath=config_yaml.filepath,
-        ) from e
+        context = FileContext(file_name=ctx.filename, line_number=ctx.start_line) if ctx is not None else None
+        issues.append(ValidationError(context=context, message=str(e)))
```

Comment:
This obliterates the stack traces, which will make debugging really painful for us if we ever hit a model validation issue that resolves to an issue in the parsing/validation stack itself.

Let's either store the stack traces or else collect exceptions and re-throw in the end.